lib/rules/no-duplicate-selectors/__tests__/index.js

```diff
@@ -84,6 +84,12 @@ testRule(rule, {
     message: messages.rejected("a b"),
     line: 1,
     column: 12,
+  }, {
+    code: "a { & {} }",
+    description: "duplicate caused by &-parent selector",
+    message: messages.rejected("a"),
```

Comment:
In the [CSS nesting instance](https://tabatkins.github.io/specs/css-nesting/#nest-selector) it looks like `&` is syntactic sugar for `:matches()`. So, I guess 

```
a { & {} }
```

Just becomes:

```
a {}
:matches(a) {}
```

Technically different selectors, but they do the same thing.

Anyhow, I think you're [excellent write-up of the edge-cases and inconsistencies](https://github.com/stylelint/stylelint/issues/2198#issuecomment-274276106) now makes me think we should just give any rule with just `&` as its selector a wide-berth i.e. don't resolve it and don't check it. If we go down this path, should we do it in stylelint or within `postcss-resolve-nested-selector`?